server/base/src/main/java/org/apache/accumulo/server/AbstractServer.java

```diff
@@ -73,6 +78,37 @@ protected AbstractServer(String appName, ConfigOpts opts, String[] args) {
         lmd.getIntervalMillis(context.getConfiguration()), TimeUnit.MILLISECONDS);
     ThreadPools.watchNonCriticalScheduledTask(future);
     processMetrics = new ProcessMetrics(context);
+    idleStopEnabled = getIdleStopEnabled(context.getConfiguration());
+    idleStopPeriod = getIdleStopPeriod(context.getConfiguration());
+  }
+
+  protected boolean getIdleStopEnabled(AccumuloConfiguration conf) {
+    return false;
+  }
+
+  protected long getIdleStopPeriod(AccumuloConfiguration conf) {
+    return 0L;
+  }
+
+  public boolean shouldStopDueToIdleCondition(Supplier<Boolean> idleCondition) {
+    if (!idleStopEnabled) {
+      return false;
+    }
+    boolean idle = idleCondition.get();
+    if (!idle) {
+      idlePeriodStart = 0;
+      return false;
+    } else if (idlePeriodStart == 0) {
+      idlePeriodStart = System.currentTimeMillis();
```

Comment:
Could use nanoTime to make the code resilient to changes in wall clock time on the host.

Reply:
I made this change in b2c3415.